CRAMv3.tex

```diff
@@ -477,15 +477,17 @@ \section{\textbf{Container header structure}}
 \hline
 \end{tabular}
 
-\subsection{\textbf{CRAM header in the first container}}
+\subsection{\textbf{BAM header in the first container}}
 
-The first container in the CRAM file contains the BAM header in an optionally
-compressed block. Only gzip is allowed as compression method for this block. BAM
-header is terminated with \textbackslash{}0 byte and any extra bytes in the
-block can be used to expand the BAM header. For example when updating @SQ
-records additional space may be required for the BAM header. It is recommended
-to reserve 50\% more space in the CRAM header block than it is required by the
-BAM header.
+The first container in a CRAM file contains the BAM header in an optionally
```

Comment:
Maybe something like:

> The first container in a CRAM file contains a textual header in a single block, optionally gzip compressed.
> This text header currently matches the SAM header specification.

From that point on we can refer to CRAM header instead of BAM header.  I can replace the images with tikz pic if you wish.

I say "single block" because that's how it is currently defined, but for future versions I want to amend this to "one or more blocks", or to permit blocks to be able to be more than 2GB in size.  I say this because people have already hit limits in both BAM and CRAM with *huge* headers due to millions of SQ lines.  If we keep blocks are 32-bit sizes, then we'd need to permit multiple blocks, with obvious concatenation properties to construct the header.

I guess it's such a trivially backwards compatible route (multi-blocks, not 64-bit sizes) that we could add that to CRAM 3.1 too.  If we chose that route in 3.1 then we'd also amend the next paragraph with "additional raw blocks" instead of "a second raw block".